test/rest-builder.test.js

```diff
@@ -204,4 +204,13 @@ describe('REST Request Builder', function () {
     });
 
   });
+
+  describe('invoke', function(){
+    it('should return a promise when no callback is specified', function(){
+      var builder = new RequestBuilder(require('./request-template.json'));
+      var promise = builder.invoke({p: 1, a: 100, b: false});
+      assert(promise.hasOwnProperty('then'));
+      assert(promise.hasOwnProperty('catch'));
+    });
```

Comment:
Please use the following assertions instead to be compatible with Node 0.12 or iojs:
```
assert(typeof promise['then'] === 'function');
assert(typeof promise['catch'] === 'function');
```